### PR TITLE
add category tvl and points.

### PR DESCRIPTION
**Pull request by robert-zklink**

### CODE REVIEW

File: src/tvl/tvl.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        categoryTvlMap.set(
+          category.name,
+          categoryTvlMap.get(category.name).plus(projectTvl),
+        );
+      }
+    }
+    const categoryTvl: CategoryTvl[] = [];
+    for (const [category, tvl] of categoryTvlMap) {
+      categoryTvl.push({ name: category, tvl });
+    }
```
Comment (xsteadybcgo) [inline src/tvl/tvl.service.ts:35]: concisely by using `const categoryTvl = Array.from(categoryTvlMap, ([name, tvl]) => ({ name, tvl }));`

File: src/projectCategory.config.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  {
+    name: "gamefi",
+    items: ["allspark"],
+  },
+  {
+    name: "other",
+    items: ["rubic", "interport", "orbiter", "symbiosis", "meson"],
+  },
+];
```
Comment (xsteadybcgo) [inline src/projectCategory.config.ts:22]: Let's create a config folder in the root directory to handle this configuration.

File: src/tvl/tvl.service.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    for (const category of categoryConfig) {
+      categoryTvlMap.set(category.name, new BigNumber(0));
+      for (const project of category.items) {
+        const projectTvl = projectsTvlMap.get(project) ?? new BigNumber(0);
+        categoryTvlMap.set(
+          category.name,
+          categoryTvlMap.get(category.name).plus(projectTvl),
+        );
+      }
+    }
```
Comment (xsteadybcgo) [inline src/tvl/tvl.service.ts:31]: Compared to a for loop, a functional programming approach can be more concise and expressive.

File: src/type/points.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  category: string;
+  project: string;
+  holdingPoints: number;
+  refPoints: number;
+}
+
+export interface CategoryTvl {
+  name: string;
+  tvl: BigNumber;
+}
```
Comment (xsteadybcgo) [inline src/type/points.ts:13]: These duplicates the related DTO definition. We can directly use the DTO to maintain the principle of a single source of truth.

File: src/nova/nova.balance.service.ts [hunk 5]
```diff
@@ -221,4 +223,31 @@ export class NovaBalanceService {
...
+      for (const project of category.items) {
+        const totalPoints = projectPointsMap.get(project);
+        projectCategoryPoints.push({
+          category: category.name,
+          project: project,
+          holdingPoints: totalPoints ? totalPoints : 0,
+          refPoints: 0,
+        });
+      }
+    }
```
Comment (xsteadybcgo) [inline src/nova/nova.balance.service.ts:250]: refer to suggestions on `categoryConfig`

File: src/projectCategory.config.ts (hunk 2 above)
Comment (xsteadybcgo) [reply src/projectCategory.config.ts:22]: I suggest changing the data structure to the one below. Although the above structure is categorized and looks concise, for business logic processing, you have to use a for loop to handle the name everywhere. If you change it to the structure below, the for loop can be omitted.

```Typescript
export const categoryConfig = [
  {
    category: "spotdex",
    project: "novaswap"
  },
  {
    category: "spotdex",
    project: "izumi"
  },
  ...
];
```

File: src/common/response.dto.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  @ApiProperty({
+    type: String,
+    description: "error message",
+    example: "no error",
+  })
+  public readonly errmsg: string;
+
+  @ApiProperty({
+    type: Number,
+    description: "elPoints and kelpMiles",
```
Comment (xsteadybcgo) [inline src/common/response.dto.ts:21]: Is this description correct?

**Review by xsteadybcgo — APPROVED** [review]

**Review by zkcarter — APPROVED** [review]

**Pull request merged**